docs/HeliumParamSpecs.md

```diff
@@ -0,0 +1,295 @@
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: List of movies filtered by actorid as [{}, {},..]
+    * Valid input - genre length > 3 characters and does not contain results, example = horro
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: Empty list []
+    * Invalid input - empty, genre length < 3 characters
+         * Status : 400
+         * Content-Type: text/plain
+         * Response Body: Invalid genre parameter
+
+
+* Name : pagesize
+* Type: integer
+* Parameter Validations:
+    * Valid input range containing results - [1, 1000]
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: List of movies filtered by pagesize as [{}, {},..]
+    * Valid input range not containing results - [1, 1000]
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: Empty list []
+    * Invalid input - empty should picks defaults [1, 1000]
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: Empty list []
+    * Invalid input - 100.23
+         * Status : 400
+         * Content-Type: text/plain
+         * Response Body: Invalid pagesize parameter
+
+
+* Name : pagenumber
+* Type: integer
+* Parameter Validations:
+    * Valid input range containing results - [1, 10000]
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: List of movies filtered by pagenumber as [{}, {},..]
+    * Valid input range not containing results - [1, 10000]
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: Empty list []
+    * Invalid input - empty values should pick defaults [1, 10000]
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: Empty list []
+    * Invalid input - 100.23
+         * Status : 400
+         * Content-Type: text/plain
+         * Response Body: Invalid pagenumber parameter
+
+
+## Multiple Error Scenarios (TBD)
+
+* Options for handling multiple errors scenarios:
+    * Consider simply adding each error to the errors array.
+    * Throw the first error occured.
+*   * Consider throwing a 422 Unprocessable Entity.
+
+
+### API Parameters
+
+* API /api/movies/{movieid}
+* Name : movieid
+* Type: string
+* Parameter Validations:
+    * Valid input - Wellformed movieid rule - starts with 'tt' followed by 9 digit integer converted to string. Example 'tt0114746'
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: Single movie document as JSON
+    * Invalid input - empty, movieid conforms wellformed String rules and does not exist, movieid not conforming to wellformed movieid rule example - 'tt1234', 'nm1234'
+         * Status : 404
+         * Content-Type: text/plain
+         * Response Body: movieid not found
+
+
+## Actors
+
+### API /api/actors
+
+### Query Parameters
+
```

Comment:
Since all of the actor parameters (q, pageSize, pageNumber, id) are the same as movies (except it's actorId instead of movieId) it would be cleaner to somehow consolidate so we dont have to keep copy pasting the rules to keep it consistent in 2 places, in case we make updates.

Reply:
Yes, agreed, updating this change.

Reply:
I think the /api/actors/{actorId} parameter also has the same rules as the actorId parameter for /api/movies/ searching/querying by actorId.  Can consolidate a little bit more there :)

Reply:
Yesterday Bart and I had a discussion on this, IMO it may not be a good experience to point to "actor id" query parameter of "movie" to refer to validations, hence made this as a separation section